pkg/predicate/predicate.go

```diff
@@ -51,3 +58,37 @@ func (GenerationChangedPredicate) Update(e event.UpdateEvent) bool {
 	}
 	return true
 }
+
+// Skips events that have labels matching a selector defined in watches.yaml
+func EventFilter(eventLabels map[string]string, r ResourceFilterPredicate) bool {
+
+	requirements, err := metav1.LabelSelectorAsSelector(&r.Selector)
+
+	if err != nil {
+		log.Error(nil, "Unable to retrieve requirement", err)
+		return false
+	}
+
+	if requirements.Matches(labels.Set(eventLabels)) {
+		return true
+	}
+
+	return false
+}
+
+// Predicate functions that call the EventFilter Function
+func (r ResourceFilterPredicate) Update(e event.UpdateEvent) bool {
+	return EventFilter(e.MetaNew.GetLabels(), r)
```

Comment:
With the signature above, these would turn into `r.EventFilter(e.Meta.GetLabels())`